File: app/src/main/java/z/nova/rifmobolgame/Model/LevelAdapter.java
package z.nova.rifmobolgame.Model;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import z.nova.rifmobolgame.R;

public class LevelAdapter extends RecyclerView.Adapter<LevelAdapter.ViewHolder> {
    final private ListItemClickListener mOnClickListener;

    private final LayoutInflater inflater;
    private final List<TextLevel> textLevels;


    public LevelAdapter(Context context, List<TextLevel> textLevels, ListItemClickListener listener){
        this.textLevels = textLevels;
        this.inflater = LayoutInflater.from(context);
        mOnClickListener = listener;
    }


    @NonNull
    @Override
    public LevelAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = inflater.inflate(R.layout.list_item, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull LevelAdapter.ViewHolder holder, int position) {
        TextLevel textLevel = textLevels.get(position);
        holder.nameView.setText(textLevel.getName());
    }

    @Override
    public int getItemCount() {
        return textLevels.size();
    }

    public interface ListItemClickListener {
        void onListItemClick(int clickItemIndex);
    }

    class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        TextView nameView;

        public ViewHolder(View view){
            super(view);
            nameView = view.findViewById(R.id.textView);
            view.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            int clickedPosition = getAbsoluteAdapterPosition();
            mOnClickListener.onListItemClick(clickedPosition);
        }
    }
}

File: app/src/main/java/z/nova/rifmobolgame/model/LevelAdapter.kt
package z.nova.rifmobolgame.model

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import z.nova.rifmobolgame.R

internal class LevelAdapter(
    context: Context?,
    private val textLevels: List<TextLevel>,
    listener: ListItemClickListener
) : RecyclerView.Adapter<LevelAdapter.ViewHolder>() {
    private val mOnClickListener: ListItemClickListener
    private val inflater: LayoutInflater
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = inflater.inflate(R.layout.list_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val textLevel = textLevels[position]
        holder.nameView.text = textLevel.name
    }

    override fun getItemCount(): Int {
        return textLevels.size
    }

    interface ListItemClickListener {
        fun onListItemClick(clickItemIndex: Int)
    }

    internal inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view),
        View.OnClickListener {
        var nameView: TextView
        override fun onClick(v: View) {
            val clickedPosition = absoluteAdapterPosition
            mOnClickListener.onListItemClick(clickedPosition)
        }

        init {
            nameView = view.findViewById(R.id.textView)
            view.setOnClickListener(this)
        }
    }

    init {
        inflater = LayoutInflater.from(context)
        mOnClickListener = listener
    }
}